p2p/p2p.go

```diff
@@ -196,16 +197,17 @@ func RegisterConnectionLogger(ctx context.Context, tcpNode host.Host, peerIDs []
 	quit := make(chan struct{})
 	defer close(quit)
 
-	type connType struct {
-		Peer peer.ID
-		Type string
+	type connKey struct {
+		PeerName string
+		Type     string
 	}
 
 	var (
-		peers      = make(map[peer.ID]bool)
-		events     = make(chan logEvent)
-		typeCounts = make(map[connType]int)
+		peers  = make(map[peer.ID]bool)
+		events = make(chan logEvent)
+		ticker = time.NewTicker(time.Second * 30)
 	)
+	defer ticker.Stop()
```

Comment:
Won't this line stop the ticker at the end of this function scope?

I believe this should be moved around like 255.

Reply:
good catch!